pandas/tests/tools/test_to_datetime.py

```diff
@@ -1073,9 +1073,17 @@ def test_datetime_invalid_datatype(self, arg):
         with pytest.raises(TypeError, match=msg):
             to_datetime(arg)
 
+    @pytest.mark.parametrize("errors", ["coerce", "raise", "ignore"])
+    def test_invalid_format_raises(self, errors):
+        # https://github.com/pandas-dev/pandas/issues/50255
+        with pytest.raises(
+            ValueError, match="':' is a bad directive in format 'H%:M%:S%"
+        ):
+            to_datetime(["00:00:00"], format="H%:M%:S%", errors=errors)
+
     @pytest.mark.parametrize("value", ["a", "00:01:99"])
     @pytest.mark.parametrize(
-        "format,warning", [(None, UserWarning), ("H%:M%:S%", None)]
+        "format,warning", [(None, UserWarning), ("%H:%M:%S", None)]
     )
     def test_datetime_invalid_scalar(self, value, format, warning):
```

Comment:
the test claims to be testing invalid scalars, but in reality it was erroring before the parsing even started because `"H%:M%:S%"` is an invalid format

So, I've changed the format to reflect what it looks like this test was meant to be testing, and have made a separate test above to check that invalid formats raise